server/migrations/2020-06-30-135809_remove_mat_views/up.sql

```diff
@@ -0,0 +1,576 @@
+  ELSIF (TG_OP = 'INSERT') THEN
+    insert into private_message_fast select * from private_message_view where id = NEW.id;
+  END IF;
+
+  return null;
+end $$;
+
+-- Comment
+
+create table comment_aggregates_fast as select * from comment_aggregates_view;
+alter table comment_aggregates_fast add column fast_id serial primary key;
+
+create index idx_comment_aggregates_fast_id on comment_aggregates_fast (id);
+
+create view comment_fast_view as
+select
+ac.*,
+u.id as user_id,
+coalesce(cl.score, 0) as my_vote,
+(select cf.id::boolean from community_follower cf where u.id = cf.user_id and ac.community_id = cf.community_id) as subscribed,
+(select cs.id::bool from comment_saved cs where u.id = cs.user_id and cs.comment_id = ac.id) as saved
+from user_ u
+cross join (
+  select
+  ca.*
+  from comment_aggregates_fast ca
+) ac
+left join comment_like cl on u.id = cl.user_id and ac.id = cl.comment_id
+
+union all
+
+select 
+    caf.*,
+    null as user_id, 
+    null as my_vote,
+    null as subscribed,
+    null as saved
+from comment_aggregates_fast caf
+;
+
+-- Do the reply_view referencing the comment_fast_view
+create view reply_fast_view as 
+with closereply as (
+    select 
+    c2.id, 
+    c2.creator_id as sender_id, 
+    c.creator_id as recipient_id
+    from comment c
+    inner join comment c2 on c.id = c2.parent_id
+    where c2.creator_id != c.creator_id
+    -- Do union where post is null
+    union
+    select
+    c.id,
+    c.creator_id as sender_id,
+    p.creator_id as recipient_id
+    from comment c, post p
+    where c.post_id = p.id and c.parent_id is null and c.creator_id != p.creator_id
+)
+select cv.*,
+closereply.recipient_id
+from comment_fast_view cv, closereply
+where closereply.id = cv.id
+;
+
+drop trigger refresh_comment on comment;
+
+create trigger refresh_comment
+after insert or update or delete
+on comment
+for each row
+execute procedure refresh_comment();
+
+-- Sample select
+-- select * from comment_fast_view where content = 'test_comment' and user_id is null;
+-- Sample insert 
+-- insert into comment(creator_id, post_id, content) values (2, 2, 'test_comment');
+-- Sample delete
+-- delete from comment where content like 'test_comment';
+-- Sample update
+-- update comment set removed = true where content like 'test_comment';
+create or replace function refresh_comment()
+returns trigger language plpgsql
+as $$
+begin
+  IF (TG_OP = 'DELETE') THEN
+    delete from comment_aggregates_fast where id = OLD.id;
+
+    -- Update community number of comments
+    update community_aggregates_fast as caf
+    set number_of_comments = number_of_comments - 1
+    from post as p
+    where caf.id = p.community_id and p.id = OLD.post_id;
+
+  ELSIF (TG_OP = 'UPDATE') THEN
+    delete from comment_aggregates_fast where id = OLD.id;
+    insert into comment_aggregates_fast select * from comment_aggregates_view where id = NEW.id;
```

Comment:
Hrm I don't follow, the text can update, `deleted` / `removed` can update, `read` can update, lots of things. Overall it just looks like a ~6ms cost to select on the specific id anyway.

Reply:
Generally from what I gathered, we're using aggregate tables to get something like, total up or downvotes on a post or comment, and how many child comments does this comment or post have.  In this context, what value does editing a comment provide to the aggregates?  It would have the same up or downvote statistics and the same number of sub-comments, so this refresh would trigger and yield the same end result